### PR TITLE
receipt loader

### PR DESCRIPTION
**Pull request by Alexey-N-Chernyshov**
Signed-off-by: turuslan <turuslan.devbox@gmail.com>

### Description of the Change
Removes cache from message loader and adds receipts loader to search message receipts on chain.

### CODE REVIEW

File: core/storage/chain/receipt_loader.hpp [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+#pragma once
+
+#include "fwd.hpp"
+#include "primitives/tipset/tipset.hpp"
+#include "vm/runtime/runtime_types.hpp"
+
+namespace fc::storage::blockchain {
+  using fc::primitives::tipset::Height;
+  using fc::vm::runtime::MessageReceipt;
```
Comment (wer1st) [inline core/storage/chain/receipt_loader.hpp:14]:
```suggestion
  using primitives::tipset::Height;
  using vm::runtime::MessageReceipt;
```

File: core/api/full_node/make.cpp [hunk 2]
```diff
@@ -928,6 +930,21 @@ namespace fc::api {
     api->StateSearchMsg = {};
     api->StateWaitMsg =
         waitCb<MsgWait>([=](auto &&cid, auto &&confidence, auto &&cb) {
+          // look for message on chain
+          auto receipt_loader =
```
Comment (wer1st) [inline core/api/full_node/make.cpp:934]:
```suggestion
          const auto receipt_loader =
```

File: core/storage/chain/receipt_loader.cpp [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+      adt::Array<MessageReceipt> receipts{tipset->getParentMessageReceipts(),
+                                          ipld_};
+      OUTCOME_TRY(parent, ts_load_->load(tipset->getParents()));
+      OUTCOME_TRY(
+          parent->visitMessages({ipld_, true, false},
+                                [&](auto i, auto, auto &msg_cid, auto, auto)
+                                    -> outcome::result<void> {
+                                  if (msg_cid == message_cid) {
+                                    OUTCOME_TRYA(found, receipts.get(i));
+                                  }
```
Comment (wer1st) [inline core/storage/chain/receipt_loader.cpp:37]: Do we need to get first or last receipt from receipts which this cid?

Comment (turuslan) [reply core/storage/chain/receipt_loader.cpp:37]: visited messages are unique

Comment (wer1st) [reply core/storage/chain/receipt_loader.cpp:37]: Yes, we already discussed it and found solution

**Review by turuslan — APPROVED** [review]

**Review by wer1st — APPROVED** [review]

**Pull request merged**